addons/account/models/account_move_line.py

```diff
@@ -700,9 +700,10 @@ def _compute_same_currency(self):
     @api.depends('currency_rate', 'balance')
     def _compute_amount_currency(self):
         for line in self:
-            if line.amount_currency is False:
+            if line.amount_currency is False or line._origin.currency_id == line.company_id.currency_id:
```

Comment:
In the first glance, `currency_id` should be correctly computed based the move's currency since it's a computed field so, imo, `_origin` should be necessary here. Then, you will end up with the same condition as the one below so:
```py
if line.currency_id == line.company_id.currency_id:
    line.amount_currency = line.balance
elif line.amount_currency is False:
    line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
```

... I guess it's not so simple :see_no_evil: